### PR TITLE
log sanitization warnings for levelHtml to new relic as custom events [finishes #112187121]

### PR DESCRIPTION
**Pull request by davidsbailey**
https://www.pivotaltracker.com/story/show/112187121

Might as well implement this now if we want to see what effect sanitizing levelHtml will have on existing applab apps.

@Bjvanminnen any tips on how to test this locally based on how you validated https://github.com/code-dot-org/code-dot-org/pull/5469 ? alternately I can patch it into production-console and test it there.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
nice....looks like this was pretty easy to add :)

lgtm. i dont remember coming up with any great testing strategy. i think i just went over to prod and copy/pasted some code into the console to make sure things worked as expected.

**Comment by davidsbailey** [comment]
Good call. I discovered a 4096 character limit on fields in the value hash and have added some truncation logic. If a different caller of logToCould provides a field longer than 4096, that field shows up empty and the rest of the log entry is unaffected.

File: apps/src/applab/designMode.js [hunk 1]
```diff
@@ -506,6 +507,11 @@ designMode.parseFromLevelHtml = function(rootEl, allowDragging, prefix) {
     var msg = "The following lines of HTML were modified or removed:\n" + removed +
       "\noriginal html:\n" + unsafe + "\nmodified html:\n" + safe;
     console.log(msg);
+    logToCloud.addPageAction(logToCloud.PageAction.SanitizedLevelHtml, {
+      removedHtml: removed.substring(0, logToCloud.MAX_LENGTH),
+      unsafeHtml: unsafe.substring(0, logToCloud.MAX_LENGTH),
+      safeHtml: safe.substring(0, logToCloud.MAX_LENGTH)
+    });
```
Comment (Bjvanminnen) [inline apps/src/applab/designMode.js:514]: I wonder if we should have addPageAction just iterate through each of the fields, and truncate them if need be.

Comment (davidsbailey) [inline apps/src/applab/designMode.js:514]: oops, sorry, I wasn't sure if you wanted to take another look. That sounds like a good improvement. I can do another PR for that.

Comment (davidsbailey) [inline apps/src/applab/designMode.js:514]: https://github.com/code-dot-org/code-dot-org/pull/6655

**Pull request merged**